airflow/sensors/external_task.py

```diff
@@ -140,21 +140,30 @@ def __init__(
                 f"`{self.allowed_states}` and failed states `{self.failed_states}`"
             )
 
-        if external_task_id is not None and external_task_ids is not None:
+        if external_task_id and external_task_ids:
             raise ValueError(
                 "Only one of `external_task_id` or `external_task_ids` may "
-                "be provided to ExternalTaskSensor; not both."
+                "be provided to ExternalTaskSensor; "
+                "Use external_task_id or external_task_ids or external_task_group_id."
             )
 
-        if external_task_id is not None:
-            external_task_ids = [external_task_id]
+        if external_task_group_id and external_task_id:
+            raise ValueError(
+                "Only one of `external_task_group_id` or `external_task_id` may "
+                "be provided to ExternalTaskSensor; "
+                "Use external_task_id or external_task_ids or external_task_group_id."
+            )
 
         if external_task_group_id and external_task_ids:
             raise ValueError(
-                "Values for `external_task_group_id` and `external_task_id` or `external_task_ids` "
-                "can't be set at the same time"
+                "Only one of `external_task_group_id` or `external_task_ids` may "
+                "be provided to ExternalTaskSensor; "
+                "Use external_task_id or external_task_ids or external_task_group_id."
             )
```

Comment:
I’m not quite sure these changes are needed, or even beneficial at all.

Reply:
Thanks for the review, added the above code to give the explicit error message between external_task_id and external_task_ids.

Reply:
The question is mainly for the `is not None` changes.

Reply:
No specific reason other than keeping all the conditions look same in this function. some are with out is not none and some are with is not none.
Are there any potential issues you foresee with this change?

Reply:
There are some weird edge cases to consider when you rely on truthiness. For example, `external_task_id=""` is an invalid input that would now fail differently, and arguably passing it through the sensor (and never actually trigger) could be a better behaviour. And `external_task_ids=[]` is arguably a valid-ish input that could be useful for dynamically generated DAGs.

Reply:
Thanks, added the is not None back.